docs/fsharp-cheatsheet.md

```diff
@@ -540,17 +538,156 @@ Another way of implementing interfaces is to use *object expressions*.
+
+    module MyNamespace.SubNamespace.Functions
+
+<a name="CodeOrganization_OpenAndAutoOpen"></a>
+### Open and AutoOpen
+
+The `open` keyword can be used with `module`, `namespace`, and `type`.
+
+    open System.Diagnostics  // open namespace
+    let stopwatch = Stopwatch.StartNew()
+    ---
+    module MyModule =
+        type DU1 = A | B | C
+        type DU2 = D | E | F
+
+    open type MyModule.DU1
+    let du1 = A
+    let duNotDefined = D  // 'D' not defined
+    open MyModule
+    let du2 = D
+
+Available to `module` elements, is the `AutoOpen` attribute. This alleviates the need for an `open`; *however* this should be used cautiously,
+as all following declarations will be immediately brought into the global namespace and cause conflicts.
+
+    [<AutoOpen>]
+    module MyModule =
+        type DU = A | B | C
+
+    let du = A
+
+### Accessibility Modifiers
+
+F# supports `public`, `private` (restraining the element to its containing `type` or `module`) and `internal` (restraining the element to its containing assembly).
+It can be applied to `module`, `let`, `member`, `type`, and `new`.
+
+With the exception of `let` bindings in a class `type`, everything defaults to `public`.
+
+| Element                                                           | Example with Modifier                      |
+|-------------------------------------------------------------------|--------------------------------------------|
+| Module                                                            | `module internal MyModule =`               |
+| Module .. `let`                                                   | `let private value =`                      |
+| Record                                                            | `type internal MyRecord = { id: int }`     |
+| Record [ctor](#CodeOrganization_PrivateConstructors)              | `type MyRecord = private { id: int }`      |
+| Discriminated Union                                               | `type private MyDiscUni = A \| B`          |
+| Discriminated Union [ctor](#CodeOrganization_PrivateConstructors) | `type MyDiscUni = internal A \| B `        |
+| Class                                                             | `type internal MyClass() =`                |
+| Class [ctor](#CodeOrganization_PrivateConstructors)               | `type MyClass private () =`                |
+| Class Additional [ctor](#CodeOrganization_PrivateConstructors)    | `internal new() = MyClass("defaultValue")` |
+| Class .. `let`                                                    | *Always private. Cannot be modified*       |
+| `type` .. `member`                                                | `member private _.classMember =`           |
+
+<a name="CodeOrganization_PrivateConstructors"></a>
+##### Private Constructors
+
+Limiting `type` constructors (ctor) accessibility is a good way to enforce value integrity.
```

Comment:
maybe this should bring in that there are primary constructors and auxillary ones (`new()`) ?
Not sure about ctor but I guess it makes sense to slip it into a cheatsheet

Reply:
I created an Issue for myself to add `new` to the class section and then create the link back. For now, I'll have to point at MS Learn.

I'll update the text, but I hesitate to add a link because the F# for Fun and Profit blogpost doesn't actually use the keyword `private`. And, interestingly, his post on single case unions says to use naming conventions to indicate a private members, rather than just using `private` (which is mentioned in the comments and updated in his book). I tried a quick web search but didn't get anything I thought was satisfying.

Reply:
That's fair - yes, there's a lot of stuff that's pretty involved and being judicious in whether to link to somethig general is definitely an important consideration

Reply:
I found the content that I think ScottW was pointing at; a YouTube of Yaron Minsky. I'll put that for now.